src/gamelogic/gamemodel/Environment.cpp

```diff
@@ -8,33 +8,46 @@
 #include "Environment.hpp"
 
 #include <utility>
+#include <stdexcept>
+#include <random>
 
 namespace GameModel {
 
-    Environment::Environment(std::shared_ptr<std::vector<Player>> players) :
-        board(std::make_shared<Board>()), players(std::move(players)) {
+    Environment::Environment(std::vector<Player> &players) : players(players) {
+        shuffleCardPile();
     }
 
-    auto Environment::getPlayers() const -> std::shared_ptr<std::vector<Player>> {
-        return players;
-    }
+    auto Environment::drawNCards(const std::size_t number) -> CardRange {
+        std::size_t drawableCards = board.getCardPile().size() + board.getDiscardPile().size();
+        if (number > drawableCards) {
+            throw std::runtime_error(std::string("Cannot create card range of size ") + std::to_string(number)
```

Comment:
Die `std::string` Konstruktoren kannst du auch weglassen

Reply:
Das sind keine Konstruktoren sondern to_string casts?

Reply:
Der Move Konstruktor ist doch in Zeile 40?

Reply:
> Das sind keine Konstruktoren sondern to_string casts?

Um den ersten const char* Literal hast du noch einen CTor

Reply:
> Der Move Konstruktor ist doch in Zeile 40?

Ich nehme an du beziehst dich auch die CardRange.hpp? Das ist nur der Move Assignment Operator

Reply:
Ja genau in der CardRange hpp in Zeile 38 mittlerweile ist doch der Konstruktor und nicht operator?